src/core/processing/qgsprocessingparameteralignrasterlayers.cpp

```diff
@@ -0,0 +1,250 @@
+{
+  switch ( outputType )
+  {
+    case QgsProcessing::PythonQgsProcessingAlgorithmSubclass:
+    {
+      QString code = QStringLiteral( "QgsProcessingParameterAlignRasterLayers('%1', %2)" )
+                     .arg( name(), QgsProcessingUtils::stringToPythonLiteral( description() ) );
+      return code;
+    }
+  }
+  return QString();
+}
+
+QString QgsProcessingParameterAlignRasterLayers::valueAsString( const QVariant &value, QgsProcessingContext &context, bool &ok ) const
+{
+  return valueAsStringPrivate( value, context, ok, ValueAsStringFlag::AllowMapLayerValues );
+}
+
+QVariant QgsProcessingParameterAlignRasterLayers::valueAsJsonObject( const QVariant &value, QgsProcessingContext &context ) const
+{
+  return valueAsJsonObjectPrivate( value, context, ValueAsStringFlag::AllowMapLayerValues );
+}
+
+QList<QgsAlignRasterData::RasterItem> QgsProcessingParameterAlignRasterLayers::parameterAsItems( const QVariant &layersVariant, QgsProcessingContext &context )
+{
+  QList<QgsAlignRasterData::RasterItem> items;
+
+  if ( qobject_cast< QgsRasterLayer * >( qvariant_cast<QObject *>( layersVariant ) ) )
+  {
+    QVariantMap vm;
+    vm["inputFile"] = layersVariant;
+    items << variantMapAsItem( vm, context );
+  }
+
+  if ( layersVariant.type() == QVariant::String )
+  {
+    QVariantMap vm;
+    vm["inputFile"] = layersVariant;
+    items << variantMapAsItem( vm, context );
+  }
+  else if ( layersVariant.type() == QVariant::List )
+  {
+    const QVariantList layersVariantList = layersVariant.toList();
+    for ( const QVariant &layerItem : layersVariantList )
+    {
+      if ( layerItem.type() == QVariant::Map )
+      {
+        const QVariantMap layerVariantMap = layerItem.toMap();
+        items << variantMapAsItem( layerVariantMap, context );
+      }
+      else if ( layerItem.type() == QVariant::String )
+      {
+        QVariantMap vm;
+        vm["inputFile"] = layerItem;
+        items << variantMapAsItem( vm, context );
+      }
+    }
+  }
+  else if ( layersVariant.type() == QVariant::StringList )
+  {
+    const auto layersStringList = layersVariant.toStringList();
+    for ( const QString &layerItem : layersStringList )
+    {
+      QVariantMap vm;
+      vm["inputFile"] = layerItem;
+      items << variantMapAsItem( vm, context );
+    }
+  }
+  else if ( layersVariant.type() == QVariant::Map )
+  {
+    const QVariantMap layerVariantMap = layersVariant.toMap();
+    items << variantMapAsItem( layerVariantMap, context );
+  }
+
+  return items;
+}
+
+QgsAlignRasterData::RasterItem QgsProcessingParameterAlignRasterLayers::variantMapAsItem( const QVariantMap &layerVariantMap, QgsProcessingContext &context )
+{
+  const QVariant layerVariant = layerVariantMap[ QStringLiteral( "inputFile" ) ];
+
+  QgsRasterLayer *inputLayer = nullptr;
+  if ( ( inputLayer = qobject_cast< QgsRasterLayer * >( qvariant_cast<QObject *>( layerVariant ) ) ) )
+  {
+    // good
```

Comment:
what are these for?

Reply:
These are safety checks, to make sure that input in the variant map is a valid raster file.

Reply:
so shouldn't we catch something in the "bad" block and return a null/invalid item ? At the moment it will result in a nullptr dereference immediately afterwards.